src/lib/common/storage.ts

```diff
@@ -0,0 +1,69 @@
+const storageQuotaErrorMessage =
+  "Unable to save because the local storage quota has been exceeded: you may need to clear out your web browser's local storage for this app, or  run in private mode before you can save again. Specific error here: ";
+
+export interface SetStorageErrorObject {
```

Comment:
We maybe don't really need this, the empty object method, or `setLocalStorageItem`. The caller in `FileManagement` could call `localStorage.setItem` directly, catch the exception, and trigger the modal when appropriate. We could export the method to summarize current local storage items, and make the modal call it again after clearing/removing an item. But up to you

Reply:
As mentioned in teams, I think it's plasuible we'd want to share this in the future and I prefer smaller files so I'd prefer to leave it this way.